python/CCLE_data.py

```diff
@@ -11,43 +11,285 @@
+    ax = sns.heatmap(data, cmap = colour)
+    ax.set(xlabel = 'Histone Markers')
+    plt.show
 
-for line in file:
-    if line.startswith('!') or line.startswith('#') or line.startswith('I') or line.startswith('^') :
-        pass
-    else:
-        ids = line.split("_")[0]
-        gene_ids.append(ids)
-file.close()
+"""
+GETTING EXPRESSION DATA
+"""
+data = pd.read_csv("./../../../CCLE_RNAseq_rsem_genes_tpm_20180929.txt", delim_whitespace=True)
+data = data.replace(0, np.nan)
+ 
+del data['transcript_ids']
+
+gene_id_list = []
+
+for gene_id in data['gene_id']:
+    gene_id = gene_id.split('.')
+    gene_id_list.append(gene_id[0])
+    
+data['gene_id'] = gene_id_list
 
 mg = mygene.MyGeneInfo()
+ginfo = mg.querymany(data['gene_id'], scopes='ensembl.gene', fields='symbol', species='human')
+
+#removed repeated element
+ginfo.pop(30544)
+
+"""
+CHANGE GENE IDs TO SYMBOLS
+"""
+
+gene = []
+for entry in ginfo:
+    if entry.get('symbol') != None:
+        gene.append(entry['symbol'])
+    else:
+        gene.append("delete")
+
+data['gene_id'] = gene
+
+data.rename(columns={'gene_id' : 'Gene'}, inplace = True)
+
+"""
+FORMATTING DATA FRAME
+"""
+
+#get list of celllines and remove tissue type
+celllines = list(data)
+celllines.pop(0)
+
+cellline_list =[]
+tissue_list = []
+for cell in celllines:
+    temp = cell.split('_')
+    cell = temp[0]
+    cellline_list.append(cell)
+    tissue = temp[1:]
+    tissue = " ".join(tissue)
+    tissue_list.append(tissue)
+
+d = {'Celllines': cellline_list, 'Tissue' : tissue_list}
+cell_df = pd.DataFrame(d)
+
+df = data
+df.columns = ['Gene'] + cellline_list
+
+temp_df = df.copy()
+
+del temp_df["Gene"]
+
+x = temp_df.values.astype(float)
+min_max_scaler = preprocessing.MinMaxScaler()
+x_scaled = min_max_scaler.fit_transform(x)
+df_normalized = pd.DataFrame(x_scaled)
+
+df_normalized.insert(0, 'Gene', df.Gene , True)
+df_normalized.columns = ['Gene'] + cellline_list
+
+#fill in NaN values
+df_normalized = df_normalized.fillna(method='ffill')
+
+
+"""
+SEARCHING FOR MATCHING GENE ALOGORITHM
+"""
+
+
+search = "10013(HDAC6) 10014(HDAC5) 3065(HDAC1) 3066(HDAC2) 51564(HDAC7) 55869(HDAC8) 79885(HDAC11) 83933(HDAC10) 8841(HDAC3) 9734(HDAC9) 9759(HDAC4) 10524(KAT5) 10724(OGA) 11143(KAT7) 124359(CDYL2) 138474(TAF1L) 1387(CREBBP) 2033(EP300) 203611(CDY2B) 23522(KAT6B) 253175(CDY1B) 2648(KAT2A) 55140(ELP3) 6872(TAF1) 7994(KAT6A) 8202(NCOA3) 84148(KAT8) 8520(HAT1) 8648(NCOA1) 8850(KAT2B) 9085(CDY1) 9329(GTF3C4) 9425(CDYL) 9426(CDY2A) 9575(CLOCK) 10919(EHMT2) 11105(PRDM7) 2145(EZH1) 2146(EZH2) 23067(SETD1B) 29072(SETD2) 387893(KMT5A) 4297(KMT2A) 51111(KMT5B) 54904(NSD3) 55870(ASH1L) 55904(KMT2E) 56979(PRDM9) 58508(KMT2C) 6419(SETMAR) 64324(NSD1) 6839(SUV39H1) 7468(NSD2) 7799(PRDM2) 79723(SUV39H2) 79813(EHMT1) 8085(KMT2D) 80854(SETD7) 83852(SETDB2) 84193(SETD3) 84444(DOT1L) 84787(KMT5C) 93166(PRDM6) 9739(SETD1A) 9757(KMT2B) 9869(SETDB1) 22992(KDM2A) 23133(PHF8) 79697(RIOX1) 79831(KDM8) 84678(KDM2B)"
```

Comment:
This is great in that it works, but imagine doing this every time we wanted to add new (and specific) genes to our query. Let's make a function that takes an input of either a file name or url, and outputs the results.